### PR TITLE
Add BART Summary

### PR DESCRIPTION
**Pull request by AntonioLopardo**
PR to add the BART summary and info on me.

### CODE REVIEW

File: _posts/2020-05-11-BART-Summary.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+<script type="text/javascript" async
+  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
+</script>
+
+![](https://firebasestorage.googleapis.com/v0/b/firescript-577a2.appspot.com/o/imgs%2Fapp%2FAntonioLprd%2FMp23o_Xx8j.jpg?alt=media&token=72dbb3c1-93a5-4b9d-8cec-3e006952568e)
+
+> Paper summary: BART: Denoising Sequence-to-Sequence Pre-training for Natural Language Generation, Translation, and Comprehension , Oct. 2019. ([link](https://arxiv.org/abs/1910.13461))
+
+## **Why is this important?**
+Lewis and et al. in this paper present valuable comparative work on different pre-training techniques and show how this kind of work can be used to guide large pre-training experiments reaching SOTA results.
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: I think it's generally a good idea to introduce abbreviations before using them. In this case I suggest replacing _SOTA_ with _state-of-the-art (SOTA)_. Makes articles like this approachable to everyone! :D

File: _posts/2020-05-11-BART-Summary.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+**Token Masking** random tokens are sampled and replaced with [MASK]<br />
+**Token Deletion** similar to masking but the token sampled are deleted and the model has to add new token in their place.<br />
+**Token Infilling** a number of text spans, i.e. contiguous group tokens, are sampled, and then they are replaced by the [MASK] token.<br />
+**Sentence Permutation** random shuffling of the document’s sentences.<br />
+**Document Rotation** a token is chosen randomly to be the start of the document, the section before the starting token is appended at the end.<br />
+
+Intuitively the techniques that work at the sentence level should help the LM learn the different roles of sentences in a paragraph or longer text and so help in NLG tasks.
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: What's a NLG task?

File: _posts/2020-05-11-BART-Summary.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+Besides the pre-training techniques, the authors also compare different LM objectives focusing on the ones used by BERT and GPT and also the ones that tried to get the best of both worlds.
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: You could replace _and also the ones_ with _as well as ones_. It is less repetitive 👍

File: _posts/2020-05-11-BART-Summary.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+**Autoregressive, left to right, LM** (GPT-2)<br />
+**Masked LM**(BERT) replace 15% of the token with [MASK] and predict the corresponding words.<br />
+**Permuted LM** (XLNet) left to right, autoregressive LM training but with the order of the words to predict chosen at random. (worth looking into more, very interesting approach)<br />
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: Is the parenthasis thought as a comment to the reader or as a reminder for you? If its the latter I think it can be removed, otherwise it might be helpful to provide some links to related readings etc.

File: _posts/2020-05-11-BART-Summary.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+![](https://lh6.googleusercontent.com/YxPMuTJc7EY2rFYIXUBVIYMjV-rloyEj2UmgJ6pbxyyMDCWzdwu4KOgRErOKJcmDe4QfC7LO-2bGE6-_0pCF1lRwJfFbjGvBbuk73oFQ8AgMdHAYDNIwDH8HlEcBI15SQKTUMIUhc_8)
+    
+From the results of this first experiments the authors draw some important conclusions.<br />
+**Token masking is crucial** Only the configurations with token masking or its variations achieve consistently great performance on different tasks.<br />
+**Left-to-right pre-training improves NLG** The Classical Language Model objective despite not doing well in inference or QA tasks achieves SOTA on ELI5(Explain Like I'm 5) a NLG benchmark that with longer sequences values coherent and grammatically correct text.<br />
+**Bidirectional encoders are crucial for QA** Ignoring future context hinders the performance of left-to-right models.<br />
+
+One Interesting note from the authors in this experiment explains how while pre-training techniques and LM objectives are important they don't tell the whole story. They report that their Permuted Language model performs much worse than XLNet due to some of the valuable architectural innovations introduced in XLNet that are not present in BART.
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: This sentence: "One Interesting note from the authors in this experiment explains how while pre-training techniques and LM objectives are important they don't tell the whole story" is a bit cumbersome to read. I would suggest restructuring a bit. 

Suggestion: "While pre-training techniques and LM objectives are important, the authors make note of the fact that they do not provide the full picture."

File: _posts/2020-05-11-BART-Summary.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    
+### **Results of the Large Scale Pre-training experiment**
+After the comparative experiment, the authors trained a bigger model with 12 layers, transformer-based architecture for autoencoding, and similar hyperparameters to RoBERTa. They used both a form of token masking at 30% and sentence permutation as pre-training text-noising techniques and run the model on 160GB of news, books, stories, and web text, again similarly  to RoBERTa
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: Double spaces at the end

File: _posts/2020-05-11-BART-Summary.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+![](https://lh4.googleusercontent.com/grHkzO3a2zUtdo-_AkH3sGhWmfGQ9_n4bAD0wm78kzVLpuzYFFdSwycnLIevdGUb5jVMJpGdA46LvZN_k0PDrCCObljSvgfcTo6PHevfpa5ZonMXn-C5tEXsUW1V33akbAIINi7whkA)
+
+BART proves especially capable in abstractive summarization talks especially in the **XSum** benchmark that as oppose to the CNN/DailyMail data-set contains very few examples of summaries where phases are present both in the summary and the original text. Besides surpassing the previous best systems in summarization by a considerable margin BART does well also in NLI tasks and QA where it is on par with the SOTA.
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: Typo: "Talks" should probably be "Tasks.
First sentence of this paragraph is a bit long. Try to reformulate

File: _posts/2020-05-11-BART-Summary.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+After the comparative experiment, the authors trained a bigger model with 12 layers, transformer-based architecture for autoencoding, and similar hyperparameters to RoBERTa. They used both a form of token masking at 30% and sentence permutation as pre-training text-noising techniques and run the model on 160GB of news, books, stories, and web text, again similarly  to RoBERTa
+![](https://lh4.googleusercontent.com/grHkzO3a2zUtdo-_AkH3sGhWmfGQ9_n4bAD0wm78kzVLpuzYFFdSwycnLIevdGUb5jVMJpGdA46LvZN_k0PDrCCObljSvgfcTo6PHevfpa5ZonMXn-C5tEXsUW1V33akbAIINi7whkA)
+
+BART proves especially capable in abstractive summarization talks especially in the **XSum** benchmark that as oppose to the CNN/DailyMail data-set contains very few examples of summaries where phases are present both in the summary and the original text. Besides surpassing the previous best systems in summarization by a considerable margin BART does well also in NLI tasks and QA where it is on par with the SOTA. 
+
+### **Qualitative Analysis**
+The paper features also examples of the summaries produced by BART that can really give a sense of how well it does on the XSum data-set.
```
Comment (Viktor2k) [inline _posts/2020-05-11-BART-Summary.md]: "The paper also features..." (word order)
"... by BART that really give a sense..." (remove "can")

**Review by Viktor2k — COMMENTED** [review]
Great overview of BART! I've posted some comments and suggestions on how we can improve on it further 👍

**Comment by omarsar** [comment]
@AntonioLopardo This is nicely done, Antonio. I added a few edits here and there but overall I like the summary and the idea it captures. 

@Viktor2k This PR is ready to be merged.

**Comment by Viktor2k** [comment]
@omarsar I might be missing something, but it does not seem like I am able to merge stuff. 
see this image

**Comment by omarsar** [comment]
@Viktor2k I had sent you an invite to become a maintainer. I think you need to accept that invite first before you are able to merge PRs.
